rcore/flash.c

```diff
@@ -117,34 +117,56 @@ struct page_hdr {
 /* flag values are *cleared* to mean "true" */
 #define FLASHFLAG(val, flag) (((val) & (flag)) == 0)
 
-static void _fs_read_page_hdr(int pg, struct page_hdr *p) {
-    flash_read_bytes(REGION_FS_START + pg * REGION_FS_PAGE_SIZE, (uint8_t *)p, sizeof(struct page_hdr));
+#define MAX_FILENAME_LEN 64
+
+struct page_hdr_with_name {
+    struct page_hdr hdr;
+    char name[MAX_FILENAME_LEN + 1];
+};
+
+static void _fs_read_page_hdr(int pg, struct page_hdr_with_name *p) {
+    flash_read_bytes(REGION_FS_START + pg * REGION_FS_PAGE_SIZE, (uint8_t *)p, sizeof(struct page_hdr_with_name));
+    p->name[(MAX_FILENAME_LEN < p->hdr.filename_len) ? MAX_FILENAME_LEN : p->hdr.filename_len] = 0;
 }
 
 static void _fs_read_page_ofs(int pg, size_t ofs, void *p, size_t n) {
     flash_read_bytes(REGION_FS_START + pg * REGION_FS_PAGE_SIZE + ofs, (uint8_t *)p, n);
 }
 
-
 static uint8_t _fs_valid = 1;
 
+typedef enum {
+    PageStateUnallocated = 0,
+    PageStateFileStart = 1,
+    PageStateFileCont = 2,
+    PageStateInvalid = 3
+} PageState;
+
+typedef struct PageFlags {
+    PageState state : 2;
+} PageFlags;
+
+static PageFlags page_flags[REGION_FS_N_PAGES];
```

Comment:
I think local style is lowercase; `struct pageflags` and `typedef enum pagestate { ... } pagestate_t;`